### ISSUE TITLE
how to train dataset with labels for V1 only (soccer-ball, substitution, cards)  (CALF model) 

### ISSUE THREAD

**Issue by Al-Hussein-96** [SilvioGiancola/SoccerNetv2-DevKit#8]

**Comment by SilvioGiancola** [comment]
Hi @Al-Hussein-96 ,

Note that SoccerNet V1 is deprecated, and we recommend you to focus on the 17 classes of the V2. You can train on the 17 classes of the V2 and use your model only for 3 classes, discarding the results for the other classes. Also, feel free to use the original implementation of CALF on the V1 available here: https://github.com/cioppaanthony/context-aware-loss .

That being said, you should be able to train for the V1 with only a few changes on in this pytorch implementation :
 - change the label files from `Labels-v2.json`  to `Labels.json`,
 - update the cfg/classes.py with the 3 classes instead of the 17 classes. See the original implementation of CALF for the values of Ks,
 - make sure the `model.py` outputs the probability for the 3 classes only instead of the 17. In the code, look for any reference of the number of classes and make sure they only return 3 instead of 17.

Good luck with your re-implementation.

**Comment by Al-Hussein-96** [comment]
@SilvioGiancola 
thank you for your response,
yes I'm trying to train with labels.json on colab, But the ram is not enough for training
Can you tell me the minimum requirements?
if i buy colab pro can i train the features on model V1 or V2 ?

**Comment by SilvioGiancola** [comment]
You'll need at least 64GB of RAM for training, because the code is loading all the features first.

**Comment by Al-Hussein-96** [comment]
> You'll need at least 64GB of RAM for training, because the code is loading all the features first.

Is there a way to reduce the use of RAM ? because the maximum RAM I have is 30 GB on colab pro

**Comment by SilvioGiancola** [comment]
You will have to change the pytorch dataset and dataloader to only read the features when needed (`__getitem__()`), that will involve a lot of reading and might slow down the training process.

**Comment by Al-Hussein-96** [comment]
> You will have to change the pytorch dataset and dataloader to only read the features when needed (`__getitem__()`), that will involve a lot of reading and might slow down the training process.

Ok, thanks you very much
